Issue #16: recognize and parse two-line reference links

spautz opened the pull request:
Detect and handle cases where the URL for a reference link is on the line after its label. This situation is created by using Prettier to format a Changelog file.

Discussion: https://github.com/oscarotero/keep-a-changelog/issues/16

oscarotero commented:
Nice. Could you add this to the tests?
There's a changelog file here https://github.com/oscarotero/keep-a-changelog/blob/master/test/changelog.md that is parsed and compared with the [expected result](https://github.com/oscarotero/keep-a-changelog/blob/master/test/changelog.expected.md)

Another question: if the `nextLine` is added to the token, it should be skipped in the `map` iteration, right? To avoid the same line twice.

spautz commented:
> Could you add this to the tests?

👍 Will do

> if the nextLine is added to the token, it should be skipped in the map iteration, right?

Right now it clears the next line after pulling it in for the link (using `allLines[index + 1] = '';`), so it doesn't get duplicated. That feels a bit dirty to me, but it works as expected and the output is unchanged.

spautz commented:
Updated with two tests: in addition to newlines at the bottom of `changelog.md`, I added a test that runs the whole file through Prettier and then ensures the parsed result is unchanged.

I noticed that the reference links are regenerated each time, so the newlines are not persisted like I thought -- but this seems inconsequential to me, since they're correct (and since they'll still be parsed correctly if they get re-split by Prettier later), and also because extra newlines around other elements (like headers) aren't persisted.

spautz commented:
The Travis CI failure looks unrelated to me: it looks like it timed out trying to install Node

oscarotero commented:
Thanks!
